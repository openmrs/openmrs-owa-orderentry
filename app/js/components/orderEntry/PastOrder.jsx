import React from 'react';
import { format } from 'date-fns';
import PropTypes from 'prop-types';

export const PastOrder = (props) => {
  const {
    drug, dosingType, action, auditInfo, dose, doseUnits, frequency,
    route, duration, durationUnits, dosingInstructions,
    quantityUnits, autoExpireDate, quantity,
  } = props;

  let details;
  if (dosingType === 'org.openmrs.SimpleDosingInstructions') {
    details = (
      <p>
        {drug.display}:
        {` ${dose}`}
        {` ${doseUnits.display},`}
        {` ${frequency.display},`}
        {` ${route.display}, for `}
        {duration && ` ${duration}`}
        {durationUnits && ` ${durationUnits.display}`}
        {dosingInstructions && ` (${dosingInstructions})`}
        {(dosingInstructions && quantityUnits) && ` (Dispense: ${quantity} ${quantityUnits.display})`}
      </p>
    );
  } else {
    details = (
      <p>
        {drug.display}:
        {dosingInstructions && ` ${dosingInstructions}`}
        {(dosingInstructions && quantityUnits) && ` (Dispense: ${quantity} ${quantityUnits.display})`}

      </p>
    );
  }
  return (
    <tr>
      <td>
        {format(auditInfo.dateCreated, 'DD-MMM-YYYY HH:MM')} {autoExpireDate && `- ${format(autoExpireDate, 'DD-MMM-YYYY HH:MM')}`}
      </td>
      <td>
        {details}
      </td>
    </tr>
  );
};

PastOrder.propTypes = {
  drug: PropTypes.shape({}).isRequired,
  action: PropTypes.string.isRequired,
  auditInfo: PropTypes.shape({}).isRequired,
  dosingType: PropTypes.string,
  dose: PropTypes.number,
  doseUnits: PropTypes.shape({}),
  frequency: PropTypes.shape({}),
  route: PropTypes.shape({}),
  duration: PropTypes.number,
  durationUnits: PropTypes.shape({}),
  dosingInstructions: PropTypes.string,
  quantityUnits: PropTypes.shape({}),
  autoExpireDate: PropTypes.string,
  quantity: PropTypes.number,
};

PastOrder.defaultProps = {
  dosingType: '',
  dose: 0,
  doseUnits: null,
  frequency: null,
  route: null,
  duration: 0,
  durationUnits: null,
  dosingInstructions: '',
  quantityUnits: null,
  autoExpireDate: '',
  quantity: null,
};

export default PastOrder;
