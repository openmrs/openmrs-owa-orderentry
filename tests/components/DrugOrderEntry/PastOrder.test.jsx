import React from 'react';
import PastOrder from '../../../app/js/components/drugOrderEntry/PastOrder';

const { drug, dateActivated } = mockData;
const props = {
  dosingType: 'org.openmrs.SimpleDosingInstructions',
  drug: { display: 'new drug' },
  doseUnits: { display: 'drops' },
  frequency: { display: 'twice a week' },
  route: { display: 'reactally' },
  quantityUnits: { display: 'drops' },
  durationUnits: { display: 'weeks' },
  dateActivated: '2018-May-20 10:59',
  dateStopped: '2018-May-20 10:59',
  autoExpireDate: '2018-May-20 10:59',
  action: '',
  dateFormat: '',
}

describe('<PastOrder />', () => {
  it('should render component', () => {
    const wrapper = shallow(<PastOrder {...props}/>);
    expect(wrapper).toMatchSnapshot()
  });

  it('should render details', () => {
    const props = {
      dosingType: '',
      drug: { display: 'new drug' },
      doseUnits: { display: 'drops' },
      frequency: { display: 'twice a week' },
      route: { display: 'reactally' },
      quantityUnits: { display: 'drops' },
      durationUnits: { display: 'weeks' },
      dateActivated: '2018-May-20 10:59',
      dateStopped: '2018-May-20 10:59',
      autoExpireDate: '2018-May-20 10:59',
      dateFormat: '',
      action: '',
    }

    const wrapper = shallow(<PastOrder {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
