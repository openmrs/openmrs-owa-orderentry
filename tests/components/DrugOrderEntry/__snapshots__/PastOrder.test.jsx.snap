// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PastOrder /> should render component 1`] = `
<tr>
  <td>
    2018-05-20T10:59:00.000+01:00
    - 2018-05-20T10:59:00.000+01:00
  </td>
  <td>
    <p>
      new drug
      :
       0
       drops,
       twice a week,
       reactally, for 
      0
       weeks
    </p>
  </td>
</tr>
`;

exports[`<PastOrder /> should render details 1`] = `
<tr>
  <td>
    2018-05-20T10:59:00.000+01:00
    - 2018-05-20T10:59:00.000+01:00
  </td>
  <td>
    <p>
      new drug
      :
    </p>
  </td>
</tr>
`;
